### PR TITLE
WIP Fix/attachmentnotfound - files are not properly deleted

### PR DESCRIPTION
**Pull request by NinaHerrmann**

### CODE REVIEW

**Comment by pejhab** [comment]
Thanks for following this up!
Shouldn't this be an emergency patch as it breaks a core feature which makes a lot of forums unusable?

**Comment by NinaHerrmann** [comment]
Sorry to disappoint you, but if we are working in the same week, it is a priority...

**Comment by pejhab** [comment]
I am not disappointed. I trust you. :-)

File: locallib.php [hunk 1]
```diff
@@ -1850,6 +1850,13 @@ function moodleoverflow_delete_post($post, $deletechildren, $cm, $moodleoverflow
             // Delete the read records.
             \mod_moodleoverflow\readtracking::moodleoverflow_delete_read_records(-1, $post->id);
 
+            // Delete the attachments.
+            try {
```
Comment (NinaHerrmann) [inline locallib.php:1854]: in this way you delete the db enrty, but NOT the actual file in the folder (db is just for where the file can be found) 
Change to somthing like 
```
$fs = get_file_storage();

// Prepare file record object
$fileinfo = array(
    'component' => 'mod_mymodule',
    'filearea' => 'myarea',     // usually = table name
    'itemid' => 0,               // usually = ID of row in table
    'contextid' => $context->id, // ID of context
    'filepath' => '/',           // any path beginning and ending in /
    'filename' => 'myfile.txt'); // any filename

// Get file
$file = $fs->get_file($fileinfo['contextid'], $fileinfo['component'], $fileinfo['filearea'], 
        $fileinfo['itemid'], $fileinfo['filepath'], $fileinfo['filename']);

// Delete it if it exists
if ($file) {
    $file->delete();
}
````
(https://docs.moodle.org/dev/File_API#Delete_file)

**Pull request merged**